src/connection-manager/connections.ts

```diff
@@ -156,18 +182,15 @@ async function handleWsMessageSubscribeTypes(
   } else if (data.type === 'manifestReceived') {
     void handleManifest(data as StreamManifest)
   } else if (data.type.includes('ledger')) {
-    const current_ledger = data as StreamLedger
-    ledger_hashes.push(current_ledger.ledger_hash)
-    if (networks) {
-      const fee: Fee = {
-        fee_base: current_ledger.fee_base,
-        reserve_base: current_ledger.reserve_base,
-        reserve_inc: current_ledger.reserve_inc,
-      }
-      network_fee.set(networks, fee)
-    }
-    if (ledger_hashes.length > LEDGER_HASHES_SIZE) {
-      ledger_hashes.shift()
+    void handleLedger(data as StreamLedger, ledger_hashes, networks)
+    // eslint-disable-next-line no-prototype-builtins -- Safeguards against some strange streams data.
+  } else if (!data.hasOwnProperty('id')) {
```

Comment:
```suggestion
  } else if (Object.prototype.hasOwnProperty.call(data, 'id')) {
```

This will allow you to get rid of the linter exception.